rasa/nlu/components.py

```diff
@@ -86,17 +86,84 @@ def validate_arguments(
 def validate_required_components_from_data(
     pipeline: List["Component"], data: TrainingData
 ):
+    # Check for entity examples but no CRFEntityExtractor
+    if data.entity_examples and not any(
+        [component.name == "CRFEntityExtractor" for component in pipeline]
+    ):
+        warnings.warn(
+            "You have defined training data consisting of entity examples, but "
+            "your NLU pipeline does not include a CRFEntityExtractor. "
+            "To extract entity examples, add a CRFEntityExtractor to your pipeline."
+        )
 
-    response_selector_exists = False
-    for component in pipeline:
-        # check if a response selector is part of NLU pipeline
-        if MESSAGE_RESPONSE_ATTRIBUTE in component.provides:
-            response_selector_exists = True
+    # Check for Regex data but RegexFeaturizer not enabled
+    if data.regex_features and not any(
+        [component.name == "RegexFeaturizer" for component in pipeline]
+    ):
+        warnings.warn(
+            "You have defined training data with regexes, but "
+            "your NLU pipeline does not include an RegexFeaturizer. "
+            "To featurize regexes for entity extraction, you need to have RegexFeaturizer in your pipeline."
+        )
+
+    # Lookup Tables config verification
+    if data.lookup_tables:
+        crf_components = [
+            component
+            for component in pipeline
+            if component.name == "CRFEntityExtractor"
+        ]
+        if not crf_components:
+            warnings.warn(
+                "You have defined training data consisting of lookup tables, but "
+                "your NLU pipeline does not include a CRFEntityExtractor. "
+                "To featurize lookup tables, add a CRFEntityExtractor to your pipeline."
+            )
+        else:
```

Comment:
let's add comments here and line 116 to clarify the 2 cases

Reply:
yep I'll eventually get back to this hopefully sometime this week or next.

Reply:
no worries, was just checking in on my stuff as well, wasn't trying to push you

Reply:
@erohmensing not sure what this refers to  - is it resolved?

Reply:
yeah just needed some more comments to explain what's happening. on it